[Hackari] iP

Hackari opened the pull request:
# MSRainy

> “I'm not going to conquer anything. The pirate king is just the freest man on the sea!” – Monkey D. Luffy[(source)](https://en.wikipedia.org/wiki/Monkey_D._Luffy)

MSRainy frees your mind from having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ **SUPER** FAST to use

All you need to do is,

1. [Download it from here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you! 😏

And it is **FREE!**

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Msrainy("./data/tasks.txt").run();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Hackari committed:
Level 0. Rename, Greet, Exit

Hackari committed:
Level 1. Echo

Hackari committed:
Level 2. Add, List

Hackari committed:
Level 3. Mark as Done

Hackari committed:
Level 4. ToDos, Events, Deadlines

Hackari committed:
Level 4. but actually committed

Hackari committed:
A-TextUiTesting Automated Text UI Testing

Hackari committed:
Level 5. Handle Errors

Hackari committed:
Level 6. Delete

Hackari committed:
Add support for saving tasks to drive

This comes in the form of the populate() and
update() function. The tasks also have a new method
toData() to store themselves as data.

Hackari committed:
Add basic functionality of accepting and parsing datetimes

Hackari committed:
Update tasks.txt

Hackari committed:
Merge branch-Level-8 into master

Hackari committed:
Add find functionality

Hackari committed:
Merge branch-Level-9 into master

Hackari committed:
Move static commands out of main, add some package organization

Hackari committed:
Move static commands out of main

Hackari committed:
Refactor main to one switch case

Hackari committed:
Refactor everything and organize to packages

Hackari committed:
Add packages

Hackari committed:
Fix tasks.txt

Hackari committed:
Merge branch-A-Packages into master

Hackari committed:
Merge MoreOOP to Packages

Hackari committed:
Merge Packages to master

Hackari committed:
Refactor all classes

Hackari committed:
Merge branch 'master' into branch-A-Gradle

Hackari committed:
Add Gradle support and checkstyle plugin

Hackari committed:
Merge branch 'branch-A-Gradle'

Hackari committed:
Add JUnit tests for Parser and Deadline

Hackari committed:
Add Gradle .jar file creation capability

Hackari committed:
Add Javadocs for all non-trivial classes and methods

Hackari committed:
Comply to coding standard apart from Javadocs

Hackari committed:
Resolve merge conflicts between A-JavaDoc and A-CodingStandard

Hackari committed:
Fix remaining checkstyle errors

Hackari committed:
Add nothing since Checkstyle already exists

Hackari committed:
Merge branch-A-Checkstyle to master

Hackari committed:
Add JavaFX and change from text-based interface to UI

Hackari committed:
Merge branch-Level-10 to master

jonaturn commented on .gitignore:
```diff
@@ -8,6 +8,9 @@
 src/main/resources/docs/
 
+# Jar Compilation Files
+src/main/java/msrainy/out
```
I like the idea of leaving out unnecessary files

jonaturn commented on src/main/java/msrainy/command/Command.java:
```diff
@@ -0,0 +1,25 @@
+ * Represents an executable command within the application.
+ * Subclasses must implement the {@code execute} method to define command behavior.
+ */
+public abstract class Command {
```
I like how you used abstract classes because you know a Command class object is not needed

jonaturn commented on src/main/java/msrainy/command/task/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yy HHmm")) + ")";
```
I think this format of datetime is unconventional, is this what you are comfortable with?

jonaturn commented on src/main/java/msrainy/command/task/Task.java:
```diff
@@ -0,0 +1,86 @@
+package msrainy.command.task;
```
Would it be better to take task out of the command package?

jonaturn commented on src/main/resources/view/DialogBox.fxml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
I like the effort to not do bare minimum only

jonaturn reviewed (commented):
Nicely done! Minor stuff that could have been done differently but I think it is all up to personal preference

Hackari committed:
Update code based on code quality guidelines

Hackari committed:
Merge branch-A-CodeQuality into master

Hackari committed:
Add assert to TaskList and Parser, fix Add

Hackari committed:
Fix merge conflicts between A-Assertions and master

Hackari committed:
Add help extension from category C

Hackari committed:
Add nicer UI to msrainy

Hackari committed:
Merge branch-A-BetterGUI to master

Hackari committed:
Add remark on use of AI

Hackari committed:
Add personality to msrainy

Hackari committed:
Merge branch-A-Personality to master

Hackari committed:
Polish up different parts of the entire project

There were various checkstyle and code quality errors.
This includes Javadocs for newly written methods,
as well as the names of certain variables.

This improved code readability.

Hackari committed:
Fix a bug in find, and add user guide

When a keyword is not found, find would return a empty
string instead of a special message.

The user guide has been written. It was done with the
help of AI tools, and checked by the creator.

Hackari committed:
Add standalone .jar file

Hackari committed:
Change app background to fit theme

Hackari committed:
Crop screenshot of Ui.png properly

Hackari committed:
Fix AI.md

Hackari committed:
Merge pull request #2 from Hackari/branch-PR

Fix AI.md

Hackari committed:
Fix a bug involving double spaces and parser

Hackari committed:
Merge branch 'branch-PR'

Hackari committed:
Merge branch 'master' of https://github.com/Hackari/ip

Hackari committed:
Fix error in importing bad data

Hackari committed:
Remove README.md
